config.m4

```diff
@@ -296,6 +296,18 @@ EOT
   PHP_ADD_BUILD_DIR([$ext_builddir/ext/integrations])
   PHP_ADD_INCLUDE([$ext_builddir/ext/integrations])
 
+  dnl Avoid cleaning rust artifacts with make clean (cargo is really good at detecting changes - and rust files are not dependent on php environment).
+  dnl However, for users who really want to clean, there's always make distclean, which will flatly remove the whole target/ directory.
+  AC_DEFUN([DDTRACE_GEN_GLOBAL_MAKEFILE_WRAP], [
+    pushdef([PHP_GEN_GLOBAL_MAKEFILE], [
+      popdef([PHP_GEN_GLOBAL_MAKEFILE])
+      PHP_GEN_GLOBAL_MAKEFILE
+      sed -i $([[ "$(uname)" != "Darwin" ]] || echo "''") -e '/^distclean:/a\'$'\n\t''rm -rf target/' -e '/.*\.a /{s/| xargs rm -f/! -path ".\/target\/*" | xargs rm -f/'$'\n}' Makefile
```

Comment:
About the `sed -i` thing, we should probably use @cataphract version in https://github.com/DataDog/dd-trace-php/pull/2735/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R48-R51